functional/FunctionalTest.java

```diff
@@ -3284,6 +3284,293 @@ public static void composeObject_test6() throws Exception {
     }
   }
 
+  /** Test: enableObjectLegalHold(String bucketName, String objectName, String versionId) */
+  public static void enableObjectLegalHold_test() throws Exception {
+    if (!mintEnv) {
+      System.out.println(
+          "Test: enableObjectLegalHold(String bucketName, String objectName, String versionId)");
+    }
+    long startTime = System.currentTimeMillis();
+    String bucketNameLegalHold = getRandomName();
+    String objectName = getRandomName();
+    try {
+      client.makeBucket(bucketNameLegalHold, null, true);
+      try (final InputStream is = new ContentInputStream(1 * KB)) {
+        client.putObject(bucketNameLegalHold, objectName, is, new PutObjectOptions(1 * KB, -1));
+      }
+
+      // Enable object legal hold.
+      client.enableObjectLegalHold(bucketNameLegalHold, objectName, "");
```

Comment:
For AWS S3, `versionId` should be passed i.e. this call will fail. Can you confirm?

Reply:
Modified the code in MinioClient.java to 
```
    if (versionId != null && !versionId.isEmpty()) {
      queryParamMap.put("versionId", versionId);
    }
``` 
So here in test case if we pass `""` , the versionId is not set and this test passes against S3.

Reply:
We may need to use this logic in all methods accepting `versionId` argument.